### PR TITLE
DeepAtlas tutorial

### PR DESCRIPTION
**Pull request by ebrahimebrahim**
### Description

Adds DeepAtlas tutorial.

This tutorial demonstrates the use of MONAI for simultaneous training of registration and segmentation models. The DeepAtlas approach, in which the two models serve as a source of weakly supervised learning for each other, is useful in situations where one has many unlabeled images and just a few images with segmentation labels. The notebook works with 3D images from the OASIS-1 brain MRI dataset.

### Status
**Ready**

### Checks

- [x] Notebook runs automatically `./runner.sh -p "-and -name '*deep_atlas*'"`

### CODE REVIEW

**Comment by ebrahimebrahim** [comment]
@wyli @Nic-Ma I'm excited to introduce this tutorial! It's ready for review any time. Thank you!

**Review by wyli — APPROVED** [review]
this looks great, thanks @ebrahimebrahim. also, would you be interested in contributing a short blog post to highlight the MONAI components for deepatlas? https://monai.medium.com/ (probably we can ask the paper authors for any comment/feedback, we worked with Zhenlin last summer...)

cc @kate-sann5100 @YipengHu

**Comment by ebrahimebrahim** [comment]
> this looks great, thanks

yay! :tada: 

> would you be interested in contributing a short blog post to highlight the MONAI components for deepatlas? https://monai.medium.com/ (probably we can ask the paper authors for any comment/feedback, we worked with Zhenlin last summer...)

We (Kitware) are actually working directly with the [UNC biomedical image analysis group](https://biag.cs.unc.edu/), which is where the paper originated. I wrote this tutorial as part of that collaboration, because the technique is very useful for our projects. I'm looking into writing up something, and I would be interested in possibly getting something into the MONAI blog. Thanks for bringing this to my attention!

**Comment by YipengHu** [comment]
> > this looks great, thanks
> 
> yay! 🎉
> 
> > would you be interested in contributing a short blog post to highlight the MONAI components for deepatlas? https://monai.medium.com/ (probably we can ask the paper authors for any comment/feedback, we worked with Zhenlin last summer...)
> 
> We (Kitware) are actually working directly with the [UNC biomedical image analysis group](https://biag.cs.unc.edu/), which is where the paper originated. I wrote this tutorial as part of that collaboration, because the technique is very useful for our projects. I'm looking into writing up something, and I would be interested in possibly getting something into the MONAI blog. Thanks for bringing this to my attention!

Glad to see some of the registration components @kate-sann5100 have been used and happy to help if useful :)

**Comment by ebrahimebrahim** [comment]
> Glad to see some of the registration components @kate-sann5100 have been used and happy to help if useful :)

Yes a lot of those components got a good test drive here. They work very well!

**Comment by wildphoton** [comment]
@ebrahimebrahim Thanks a lot for making a tutorial for DeepAtlas! I think in the original paper, I pretrained a registration network in an unsupervised way as before joint training. But it is even better if it works well from the randomly initialized registration net. It would be great if you could add an evaluation on the Dice score for segmentation/registration to show the metric improvement in addition to losses.

**Comment by ebrahimebrahim** [comment]
> @ebrahimebrahim Thanks a lot for making a tutorial for DeepAtlas!

Hey, thanks for commenting! I am honored that you like it :grinning: 

> I think in the original paper, I pretrained a registration network in an unsupervised way as before joint training. But it is even better if it works well from the randomly initialized registration net.

Yeah I'm really glad it worked well without pretraining the registration network, because the notebook is already a bit long for a tutorial. 

> It would be great if you could add an evaluation on the Dice score for segmentation/registration to show the metric improvement in addition to losses.

Good idea. This is essentially already done for `seg_net`; the "validation loss" displayed there is the dice score. 
I may include the `reg_net` dice score in a future version-- currently the validation loss shown for `reg_net` is just the same as the training objective function.

**Comment by wyli** [comment]
Thanks again for the comments and feedback, I'm merging this version now.
